### DISCUSSION TITLE
How to make 127501 (digital switching) support for B&G Naviop Loop-S?

### DISCUSSION THREAD

**Discussion by htool** [canboat/canboat#297] [General]
In order to emulate a Loop-S digital switch I reverse engineered the way pgn 127501 is used.

This packet doesn't carry a manufactured id etc, so I'm not sure now how I can distinguish this from the default digital switching packet. Is there a way to do so?

    {"Binary Switch Bank Status",
     127501,
     PACKET_COMPLETE,
     PACKET_SINGLE,
     8,
     0,
     {{"Instance", BYTES(1), 1, false, 0, ""},
      {"Switch1", 1, RES_LOOKUP, false, LOOKUP_OFF_ON, ""},
      {"Fuse1", 1, RES_LOOKUP, false, LOOKUP_OFF_ON, ""},
      {"Switch5", 1, RES_LOOKUP, false, LOOKUP_OFF_ON, ""},
      {"Fuse2", 1, RES_LOOKUP, false, LOOKUP_OFF_ON, ""},
      {"Switch2", 1, RES_LOOKUP, false, LOOKUP_OFF_ON, ""},
      {"Fuse3", 1, RES_LOOKUP, false, LOOKUP_OFF_ON, ""},
      {"Fuse4", 2, RES_LOOKUP, false, LOOKUP_OFF_ON, ""},
      {"Switch3", 1, RES_LOOKUP, false, LOOKUP_OFF_ON, ""},
      {"Fuse5", 1, RES_LOOKUP, false, LOOKUP_OFF_ON, ""},
      {"Fuse6", 2, RES_LOOKUP, false, LOOKUP_OFF_ON, ""},
      {"Switch4", 1, RES_LOOKUP, false, LOOKUP_OFF_ON, ""},
      {"Fuse7", 1, RES_LOOKUP, false, LOOKUP_OFF_ON, ""},
      {"Fuse8", 2, RES_LOOKUP, false, LOOKUP_OFF_ON, ""},
      {"Switch6", 1, RES_LOOKUP, false, LOOKUP_OFF_ON, ""},
      {"Fuse9", 1, RES_LOOKUP, false, LOOKUP_OFF_ON, ""},
      {"Fuse10", 2, RES_LOOKUP, false, LOOKUP_OFF_ON, ""},
      {"Fuse11", 2, RES_LOOKUP, false, LOOKUP_OFF_ON, ""},
      {"Fuse12", 2, RES_LOOKUP, false, LOOKUP_OFF_ON, ""},
      {"Switch7", 1, RES_LOOKUP, false, LOOKUP_OFF_ON, ""},
      {"Fuse13", 1, RES_LOOKUP, false, LOOKUP_OFF_ON, ""},
      {"Switch8", 1, RES_LOOKUP, false, LOOKUP_OFF_ON, ""},
      {"Fuse14", 1, RES_LOOKUP, false, LOOKUP_OFF_ON, ""},
      {"Fuse15", 2, RES_LOOKUP, false, LOOKUP_OFF_ON, ""},
      {"Fuse16", 2, RES_LOOKUP, false, LOOKUP_OFF_ON, ""},
      {"Fuse16", 2, RES_LOOKUP, false, LOOKUP_OFF_ON, ""},
      {0}}},

**Comment by keesverruijt** [comment]
This is a standard PGN, not a company specific one. So it cannot be "distinguished" from a normal one.

I don't see any difference between your version and the standard PGN other than you renaming fields. I have a feeling these names are likely to be specific to the programming of your device, and might not be the same on other Naviop devices. The ordering of switches and fuses seems random.

**Reply by htool** [reply]
It's not random and is how the Loop-S sends and the B&G MFD shows the info. 
But as it can not be distinguished from the current one, I'll work around it within the SignalK plugin

**Reply by htool** [reply]
BTW, main difference is that the bits are interpreted individually and not per pair.
So meaning is given to value 2 and 3.

**Reply by keesverruijt** [reply]
> BTW, main difference is that the bits are interpreted individually and not per pair.
> So meaning is given to value 2 and 3.

But only for some fields, according to your new definition? I would say this is a violation of the standard...

**Comment by htool** [comment]
This can be closed. Due to incompatibility I've made the SignalK plugin land it in electrical.naviop.switches.
